submodules/radios/iridium.py

```diff
@@ -0,0 +1,252 @@
+import logging
+from threading import Lock
+from functools import partial
+
+from . import Radio
+from core import ThreadHandler
+
+from serial import Serial
+
+
+class Iridium(Radio):
+    def __init__(self, config):
+        """
+        Assumes Iridium is in low power mode on start. Sets up class fields.
+        :param config: the config dictionary loaded from config_default.yml
+        """
+        self.config = config
+        self.modules = dict()
+
+        self.logger = logging.getLogger("IRIDIUM")
+        self.read_lock = Lock()
+
+        self.serial = None
+        self.processes = {
+            "listen_thread": ThreadHandler(
+                target=partial(self.listen),
+                name="iridium-listen",
+                parent_logger=self.logger,
+            )
+        }
+
+    def start(self):
+        """
+           Opens the Iridium serial port and starts the listening thread.
+           Assumes enough power is present therefore the tty port exists.
+           If the Iridium check fails, it raises an error
+        """
+
+        self.serial = Serial(
+            self.config["iridium"]["serial_port"], baudrate=19200, timeout=30
+        )
+        self.serial.flush()
+
+        if self.check(5):
+            self.logger.debug("Iridium Check Successful")
+        else:
+            raise RuntimeError("Iridium Check Failed")
+
+        self.processes["listen_thread"].start()
+
+    def enter_low_power_mode(self):
+        """
+        Enters the Iridium into low power mode.
+        Closes the serial port and pauses the listening thread
+        Assumes Iridium is in normal mode
+        """
+        self.processes["listen_thread"].pause()
+        self.serial.close()
+
+    def enter_normal_mode(self):
+        """
+        Enters the Iridium into normal mode.
+        Re-opens the serial port and resumes the listening thread
+        Assumes Iridium is in low power mode.
+        """
+
+        self.serial.open()
+        self.processes["listen_thread"].resume()
+
+    def set_modules(self, modules):
+        self.modules = modules
+
+    def has_modules(self):
+        return len(self.modules) is not 0
+
+    def write_to_serial(self, command: str) -> (str, bool):
```

Comment:
Have you tested device-specific methods yet? If not that is fine, I just want to know